src/components/utils/src/threads/async_runner.cc

```diff
@@ -88,6 +92,22 @@ void AsyncRunner::AsyncRunnerDelegate::waitForDelegate() {
   }
 }
 
+void AsyncRunner::AsyncRunnerDelegate::clearDelegateQueue() {
+  if (!delegates_queue_.empty()) {
+    std::queue<threads::ThreadDelegate*> empty_queue;
+    delegates_queue_lock_.Acquire();
+    std::swap(delegates_queue_, empty_queue);
+    delegates_queue_lock_.Release();
+    do {
+      auto delegate = empty_queue.front();
```

Comment:
@SNiukalov The name looks bad. How can we pop and delete something from an empty queue? ))

Reply:
@AByzhynar @LitvinenkoIra 
Fixed in:https://github.com/LuxoftSDL/sdl_core/pull/41/commits/ed572e6c5d598826927c006d9ae144b4b554e5ab